core/queryalgebra/evaluation/src/main/java/org/eclipse/rdf4j/query/algebra/evaluation/iterator/CrossProductIteration.java

```diff
@@ -67,4 +67,12 @@ protected BindingSet getNextElement()
 
 		return null;
 	}
+
+	@Override
+	protected void handleClose()
+		throws QueryEvaluationException
+	{
+		super.handleClose();
```

Comment:
The two actions here need to be in a try-finally configuration to ensure both actions occur when exceptions are thrown from super.handleClose.